functions/Set-DbaTempDbConfig.ps1

```diff
@@ -203,13 +203,21 @@ function Set-DbaTempDbConfig {
 
             $logfile = Get-DbaDbFile -SqlInstance $server -Database tempdb | Where-Object Type -eq 1 | Select-Object LogicalName, PhysicalName, @{L = "SizeMb"; E = { $_.Size.Megabyte } }
 
-            if ($LogFileSize) {
+            if ($LogPath -or $LogFileSize) {
                 $Filename = Split-Path $logfile.PhysicalName -Leaf
                 $LogicalName = $logfile.LogicalName
-                $NewPath = "$LogPath\$Filename"
+                
+                if ($logPath) {
+                    $NewPath = "$LogPath\$Filename"     
+                } else {
+                    Split-Path $logfile.PhysicalName
```

Comment:
I'm not sure on that one, it's how it was in the original code and it seems to work correctly. It builds the full name of the new path based on the parameter passed in $LogPath and the file name generated from:

`$Filename = Split-Path $logfile.PhysicalName -Leaf`

the 

`$logfile.PhysicalName`

returns the current log, not the ne,  path from

`$logfile = Get-DbaDbFile -SqlInstance $server -Database tempdb | Where-Object Type -eq 1 | Select-Object LogicalName, PhysicalName, @{L = "SizeMb"; E = { $_.Size.Megabyte } }`

Reply:
But `Split-Path $logfile.PhysicalName` doesn't assign a variable, it takes the filename, splits it and puts the result in the output stream. Do you have a test, where $logPath is $null - so that this line is executed?

Reply:
Ah, I see what you're saying now, sorry, I completely miss understood the section you were referencing. 

`if ($logPath) {
                    $NewPath = "$LogPath\$Filename"   `

not the 

```
} else {
                    Split-Path $logfile.PhysicalName
```

Yes, it should use the full original path if one isn't passed in. 

```
} else {
          $newPath = $logfile.PhysicalName
```

Reply:
Then please update your code, commit and push. Then we can see the changes here.

Reply:
Here's what I've used to test the different scenarios - think it's good. Committing code now.

```
$server = "TestDB"
[int]$DataFileSize
[int]$LogFileSize
[int]$DataFileGrowth = 512
[int]$LogFileGrowth = 512
[string]$DataPath
[string]$LogPath
[string]$OutFile
[switch]$OutputScriptOnly
[switch]$DisableGrowth
[switch]$EnableException


#Run this first to get current settings then compare output of sql to what's expected.
$logfile = Get-DbaDbFile -SqlInstance $server -Database tempdb | Where-Object Type -eq 1 | Select-Object LogicalName, PhysicalName, @{L = "SizeMb"; E = { $_.Size.Megabyte } }

#Pass in nothing, nothing updated
$LogPath = $null
$NewPath = $null
$sql = $null
$LogFileSize = $null

if ($LogPath -or $LogFileSize) {
    $Filename = Split-Path $logfile.PhysicalName -Leaf
    $LogicalName = $logfile.LogicalName
                
    if ($logPath) {
        $NewPath = "$LogPath\$Filename"     
    } else {
        $NewPath = $logfile.PhysicalName
    }
                
    if (-not($LogFileSize)) {
        $LogFileSize = $logfile.SizeMb
    }

    $sql += "ALTER DATABASE tempdb MODIFY FILE(name=$LogicalName,filename='$NewPath',size=$LogFileSize MB,filegrowth=$LogFileGrowth);"
}

$sql

#Pass in log file size, everything else remains the same, log file size new
cls
$logfile
$LogPath = $null
$NewPath = $null
$sql = $null
$LogFileSize = 200

if ($LogPath -or $LogFileSize) {
    $Filename = Split-Path $logfile.PhysicalName -Leaf
    $LogicalName = $logfile.LogicalName
                
    if ($logPath) {
        $NewPath = "$LogPath\$Filename"     
    } else {
        $NewPath = $logfile.PhysicalName
    }
                
    if (-not($LogFileSize)) {
        $LogFileSize = $logfile.SizeMb
    }

    $sql += "ALTER DATABASE tempdb MODIFY FILE(name=$LogicalName,filename='$NewPath',size=$LogFileSize MB,filegrowth=$LogFileGrowth);"
}

$sql

#Pass in log path, everything else remains the same, log path new
cls
$logfile
$LogPath = "T:TestMyLogPath"
$NewPath = $null
$sql = $null
$LogFileSize = $null

if ($LogPath -or $LogFileSize) {
    $Filename = Split-Path $logfile.PhysicalName -Leaf
    $LogicalName = $logfile.LogicalName
                
    if ($logPath) {
        $NewPath = "$LogPath\$Filename"     
    } else {
        $NewPath = $logfile.PhysicalName
    }
                
    if (-not($LogFileSize)) {
        $LogFileSize = $logfile.SizeMb
    }

    $sql += "ALTER DATABASE tempdb MODIFY FILE(name=$LogicalName,filename='$NewPath',size=$LogFileSize MB,filegrowth=$LogFileGrowth);"
}

$sql

#Pass in log path and log size, everything else remains the same, log path and size new
cls
$logfile
$LogPath = "T:TestMyLogPath"
$NewPath = $null
$sql = $null
$LogFileSize = 200

if ($LogPath -or $LogFileSize) {
    $Filename = Split-Path $logfile.PhysicalName -Leaf
    $LogicalName = $logfile.LogicalName
                
    if ($logPath) {
        $NewPath = "$LogPath\$Filename"     
    } else {
        $NewPath = $logfile.PhysicalName
    }
                
    if (-not($LogFileSize)) {
        $LogFileSize = $logfile.SizeMb
    }

    $sql += "ALTER DATABASE tempdb MODIFY FILE(name=$LogicalName,filename='$NewPath',size=$LogFileSize MB,filegrowth=$LogFileGrowth);"
}

$sql
```

Everything gave the expected output.

Reply:
Hi, is there anything else I need to do? Sorry, it's my first attempt at helping out :) Thanks

Reply:
Not at the moment. I'll have a closer look at the code later today.